salt/metalk8s/addons/prometheus-operator/post-cleanup.sls

```diff
@@ -0,0 +1,51 @@
+{#- For the Prometheus-operator chart, after a cluster upgrade/downgrade some
+    remnants of the old installation might remain such as PrometheusRule CR's.
+    This state is meant to cleanup old CR's marked with labels of an old
+    Metalk8s version #}
+
+{%- set version = pillar.metalk8s.cluster_version %}
+
+{#- Todo: In future, we might need to provide a complete list of object kind
+    and apiversions that require cleanup. For now, we only handle
+    PrometheusRule CR's #}
+
+{%- set metalk8s_objects = salt.metalk8s_kubernetes.list_objects(
+        kind="PrometheusRule",
+        apiVersion="monitoring.coreos.com/v1",
+        namespace="metalk8s-monitoring",
+        label_selector="app.kubernetes.io/part-of=metalk8s"
+    ) %}
+
+{%- if metalk8s_objects and metalk8s_objects|length > 0 %}
```

Comment:
```suggestion
{%- if metalk8s_objects %}
```
You do not need to check for length > 0, `bool([]) = False` anyway

Reply:
ok